src/hidimstat/base_variable_importance.py

```diff
@@ -34,104 +112,178 @@ def __init__(self):
         super().__init__()
         self.importances_ = None
         self.pvalues_ = None
-        self.selections_ = None
 
-    def selection(
-        self, k_best=None, percentile=None, threshold=None, threshold_pvalue=None
+    def _check_importance(self):
+        """
+        Checks if the importance scores and p-values have been computed.
+        """
+        if self.importances_ is None:
+            raise ValueError(
+                "The importances need to be called before calling this method"
+            )
+
+    def importance_selection(
```

Comment:
Should we already anticipate that importances could be a 2D array? 
And consequently have an aggregation strategy like 'mean', 'median', 'max' ...